nixos/modules/programs/appgate-sdp.nix

```diff
@@ -0,0 +1,19 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+{
+  options = {
+    programs.appgate-sdp = {
+      enable = mkEnableOption
+        "AppGate SDP VPN client";
+    };
+  };
+
+  config = mkIf config.programs.appgate-sdp.enable {
+    boot.kernelModules = [ "tun" ];
+    environment.systemPackages = [ pkgs.appgate-sdp ];
+    services.dbus.packages = [ pkgs.appgate-sdp ];
+    systemd.packages = [ pkgs.appgate-sdp ];
```

Comment:
Awesome! :tada: 

Unfortunately there is [bug](https://github.com/NixOS/nixpkgs/issues/81138) in NixOS... so additional to this you should add something like this:

```
systemd.services.appgate-dumb-resolver.wantedBy =  [ "multi-user.target" ];
systemd.services.appgatedriver.wantedBy =  [ "multi-user.target" ];
systemd.services.appgate-dumb-resolver.wantedBy =  [ "multi-user.target" ];
systemd.services.appgate-resolver.wantedBy =  [ "multi-user.target" ];
```

Reply:
it does work for me though, but I'm going to give it another test

Reply:
Oh... Do you mind commenting in the issue thread that you don't have that problem? More data points.

Reply:
You are 100% correct. `[Install]` section was present but the unit remained in `Inactive` state. I added suggested fix. Thanks!

Reply:
Ah ok. This module looks great!